internal/rules/resource_management.go

```diff
@@ -70,6 +98,17 @@ func NewResourceManagementServiceWithScheme(newClient client.Client, scheme *run
 	}
 }
 
+// SetGCTracker sets the garbage collection tracker for filtering delete events.
+func (r *ResourceManagementService) SetGCTracker(tracker GCTrackerInterface) {
+	r.gcTracker = tracker
+}
+
+func (r ResourceManagementService) markForDeletion(obj client.Object) {
+	if r.gcTracker != nil {
```

Comment:
When is gcTracker nil?

Reply:
The answer is in unit tests - we might make this non-nilable by setting it in the `New...`